### DISCUSSION TITLE
Cannot read property 'rowContext' of undefined

### DISCUSSION THREAD

**Discussion by lowrykun** [shlomiassaf/ngrid#181] [Q&A] [answered]
After upgrading to the latest version we're now getting this error constantly:

TypeError: Cannot read property 'rowContext' of undefined
  File "webpack:///node_modules/@pebula/ngrid/__ivy_ngcc__/fesm2015/pebula-ngrid-target-events.js", line 460, in PblNgridTargetEventsPlugin.setupDomEvents
                        event.context = root.context.rowContext;
  File "webpack:///node_modules/@pebula/ngrid/__ivy_ngcc__/fesm2015/pebula-ngrid-target-events.js", line 620, in PblNgridTargetEventsPlugin.setupDomEvents
                        const rowEvent = createRowEvent(rowTarget, source, cellEvent);

Any thoughts?

**Comment by shlomiassaf** [comment]
> After upgrading to the latest version we're now getting this error constantly:
> 
> 
> 
> TypeError: Cannot read property 'rowContext' of undefined
> 
>   File "webpack:///node_modules/@pebula/ngrid/__ivy_ngcc__/fesm2015/pebula-ngrid-target-events.js", line 460, in PblNgridTargetEventsPlugin.setupDomEvents
> 
>                         event.context = root.context.rowContext;
> 
>   File "webpack:///node_modules/@pebula/ngrid/__ivy_ngcc__/fesm2015/pebula-ngrid-target-events.js", line 620, in PblNgridTargetEventsPlugin.setupDomEvents
> 
>                         const rowEvent = createRowEvent(rowTarget, source, cellEvent);
> 
> 
> 
> Any thoughts?

I think I know the root cause 

Will fix next week

**Comment by laryk9** [comment]
@lowrykun Can you describe how to reproduce this error? did you by any chance use the "keepAlive" functionality on the data source?

**Reply by lowrykun** [reply]
I was trying to get an answer from our engineering team. I just looked and it appears the .keepAlive() is enabled.

**Reply by shlomiassaf** [reply]
Yea

It happen to us as well only with keep alive

We will fix this in the next couple of days

Btw @lowrykun, what company are you working at, I'm always intrigued who uses nGrid

**Reply by lowrykun** [reply]
Thank you.

I'm the Head of Product at [Finch](https://www.finch.com). We've been using it as part of our [Insights Reporting](https://www.finch.com/advertising-management-platform/insight-campaign-reports/) for the past 2 years or so.

**Reply by shlomiassaf** [reply]
@lowrykun We did a super quick fix here, version 3.1.4 let me know if you have issues

**Comment by shlomiassaf** [comment, marked as answer]
Fixed in 3.1.4

**Reply by lowrykun** [reply]
Thank you! I'll have our engineering team update ASAP.

**Reply by shlomiassaf** [reply]
@lowrykun I tried looking at your site, couldn't find a way to login to see the dashboard, do you provide a demo?

**Comment by lowrykun** [comment]
We use it inside our platform. If you're interested, email me at
***@***.*** and we can set up a Video Call.
